go-controller/pkg/ovn/controller/apbroute/external_controller_pod.go

```diff
@@ -405,45 +84,51 @@ func getMultusIPsFromNetworkName(pod *v1.Pod, networkName string) (sets.Set[stri
 	return nil, fmt.Errorf("unable to find multus network %s in pod %s/%s", networkName, pod.Namespace, pod.Name)
 }
 
-func (m *externalPolicyManager) filterNamespacesUsingPodGateway(key ktypes.NamespacedName) sets.Set[string] {
-	namespaces := sets.New[string]()
+func (m *externalPolicyManager) listPoliciesUsingPodGateway(key ktypes.NamespacedName) ([]*adminpolicybasedrouteapi.AdminPolicyBasedExternalRoute, error) {
+
+	ret := make([]*adminpolicybasedrouteapi.AdminPolicyBasedExternalRoute, 0)
+	policies, err := m.routeLister.List(labels.Everything())
+	if err != nil {
+		return nil, err
+	}
+	for _, p := range policies {
+		klog.V(5).InfoS("Checking for policy %s to have pod %s", p.Name, key)
+		pp, err := m.processExternalRoutePolicy(p)
+		if err != nil {
+			return nil, err
+		}
+		if _, found := pp.dynamicGateways[key]; found {
+			klog.V(5).InfoS("Policy %s has pod %s", p.Name, key)
+			ret = append(ret, p)
+		}
+
+	}
+	return ret, nil
+}
+
+func (m *externalPolicyManager) listPoliciesInNamespacesUsingPodGateway(key ktypes.NamespacedName) (sets.Set[string], error) {
```

Comment:
Maybe we implement the same type of "wait until all currently processed policies have quiesced" by iterating through all the ones we know about? I think that would at least greatly reduce the chance of it ever happening, and seems like the most simple approach.

Another idea: when a policy identifies a namespace it is adding to its policy, it could queue that namespace back to the namespace handler, which would force another requeue of the policy key :) queue add inception...